api_utils.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <memory>
+#include <fstream>
+#include <iostream>
+#include <ctime>
+#include <fcntl.h> // For ::open
+#include <limits>
+
+#include <google/protobuf/io/coded_stream.h>
+#include <google/protobuf/io/zero_copy_stream_impl.h>
+#include <google/protobuf/text_format.h>
+
+using std::cout;
+using std::endl;
+
+
+struct InferDeleter {
```

Comment:
I've seen these two functions in multiple places in the code place. Maybe put them in a common place?

Reply:
@yinghai noted. I've moved all the common functions to a common.hpp file. Please review my latest commit